pkg/reconciler/route/route.go

```diff
@@ -573,3 +575,24 @@ func setTargetsScheme(rs *v1alpha1.RouteStatus, dnsNames []string, scheme string
 		}
 	}
 }
+
+func buildIngressSplitsForChallenges(domains []string, challenges []netv1alpha1.HTTP01Challenge) []netv1alpha1.IngressRule {
+	splits := []netv1alpha1.IngressRule{}
+	for _, challenge := range challenges {
+		splits = append(splits, *resources.MakeIngressRuleWithPath(
+			domains,
+			challenge.ServiceNamespace,
+			false,
+			[]netv1alpha1.IngressBackendSplit{{
+				IngressBackend: netv1alpha1.IngressBackend{
+					ServiceNamespace: challenge.ServiceNamespace,
+					ServiceName:      challenge.ServiceName,
+					ServicePort:      challenge.ServicePort,
+				},
+				Percent: 100,
```

Comment:
If we're going over splits, and you're assigning 100% to all of them, will this validate?

Reply:
I will rename this function since it's actually creating IngressRules. For each desired cert with a matching solver service we create a separate `IngressRule` with a single `IngressBackendSplit` that points to the challenge solver service. Each cert has a distinct DNS name so there is no conflict.